osu.Game.Tests/NonVisual/Filtering/FilterMatchingTest.cs

```diff
@@ -159,6 +159,30 @@ public void TestCriteriaMatchingTerms(string terms, bool filtered)
             Assert.AreEqual(filtered, carouselItem.Filtered.Value);
         }
 
+        [Test]
+        [TestCase("\"artist\"", false)]
+        [TestCase("\"arti\"", true)]
+        [TestCase("\"artist title author\"", true)]
+        [TestCase("\"artist\" \"title\" \"author\"", false)]
+        [TestCase("\"an artist\"", true)]
+        [TestCase("\"tags too\"", false)]
+        [TestCase("\"tags to\"", true)]
+        [TestCase("\"version\"", false)]
+        [TestCase("\"an auteur\"", true)]
```

Comment:
I don't get these test cases either. They don't match what I think the discussion poster wanted the feature to be.

In my understanding, the test cases here should be:

```csharp
        [TestCase("\"artist\"", true)] // full phrase matches no full metadata field
        [TestCase("\"arti\"", true)] // full phrase matches no full metadata field
        [TestCase("\"the artist\"", false)] // full phrase matches a metadata field fully
        [TestCase("\"title goes here\"", false)] // full phrase matches no full metadata field
        [TestCase("\"the author\"", false)] // full phrase matches a metadata field fully
        [TestCase("\"the author\" artist", false)] // full phrase matches no full metadata field + non-quoted phrase matches a word in metadata
        [TestCase("\"the author\" \"artist\"", true)] // full phrase 2 matches no full metadata field (even though full phrase 1 is ok)
        [TestCase("\"the author\" \"the artist\"", true)] // full phrases match 2 metadata fields fully
        [TestCase("\"artist title author\"", true)] // full phrase matches no full metadata field
        [TestCase("\"artist\" \"title\" \"author\"", true)] // full phrases match no full metadata field
        [TestCase("\"an artist\"", true)] // full phrase matches no full metadata field
        [TestCase("\"tags\"", false)] // is a tag - tags are space-delimited
        [TestCase("\"tags too\"", true)] // shouldn't do full phrase matching on tags, which are space-delimited after all
        [TestCase("\"tags to\"", true)] // just doesn't match
        [TestCase("\"version\"", true)] // full phrase matches no full metadata field
        [TestCase("\"an auteur\"", true)] // full phrase matches no full metadata field
```

Reply:
again, depends on whether we are aiming to please one specific user or to implement a standard

Reply:
Okay, sure, in that case I'll disregard the discussion and review this again from the perspective of how this would generally work, I guess.